### PR TITLE
PLANET-7454 Implement GF variables in page load

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7454](https://jira.greenpeace.org/browse/PLANET-7454)

We want them to be available before form submission

### Testing

You can make sure that the dataLayer new values are correctly added on any page with a Gravity Forms form, such as [this one](https://www-dev.greenpeace.org/test-pluto/gravity-forms-test/) 

Screenshot from 2024-09-10 09-59-07

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [jupiter](https://www-dev.greenpeace.org/test-jupiter) | [admin](https://www-dev.greenpeace.org/test-jupiter/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-jupiter/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-jupiter) | [composer-local.json](https://github.com/greenpeace/planet4-test-jupiter/blob/main/composer-local.json)

:watch: 2024.09.05 10:03:59

File: src/GravityFormsExtensions.php [hunk 1]
```diff
@@ -646,7 +646,7 @@ public function p4_client_side_gravityforms_prefill(array $form): array
     {
         $supported_field_types = ['GF_Field_Hidden'];
 
-        $gf_fronted_populate = [];
+        $gf_frontend_populate = [];
```
Comment (mleray) [inline src/GravityFormsExtensions.php:649]: There was a small typo, `fronted` => `frontend`

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [pluto](https://www-dev.greenpeace.org/test-pluto) | [admin](https://www-dev.greenpeace.org/test-pluto/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-pluto/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-pluto) | [composer-local.json](https://github.com/greenpeace/planet4-test-pluto/blob/main/composer-local.json)

:watch: 2024.09.05 14:51:12

**Review by mardelnet — COMMENTED** [review]
@mleray 
These are like edge cases, but what happens if:
1. More than one form is added to the same page?
2. The same block is added multiple times to the same page?

File: assets/src/js/gravityforms-client-side.js [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
-    p4GfClientSideConfig.populate.forEach(field => {
-      const value = urlParams.get(field.parameter);
-
-      if (value !== null) {
-        document.getElementById(field.fieldId).value = value;
-      }
-    });
+  if (typeof p4GfClientSideConfig === 'undefined') {
+    return;
   }
```
Comment (mardelnet) [inline assets/src/js/gravityforms-client-side.js:6]: Should we also check for `dataLayer` here?

```suggestion
  if (typeof p4GfClientSideConfig === 'undefined') {
    return;
  }
  if (typeof dataLayer === 'undefined') {
    return;
  }
```

File: assets/src/js/gravityforms-client-side.js [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
+    });
+  });
+
+  const urlParams = new URLSearchParams(window.location.search);
+  // eslint-disable-next-line no-undef
+  p4GfClientSideConfig.populate.forEach(field => {
+    const value = urlParams.get(field.parameter);
+
+    if (value !== null) {
+      document.getElementById(field.fieldId).value = value;
```
Comment (mardelnet) [inline assets/src/js/gravityforms-client-side.js:21]: Should we check whether `field.fieldId` is not falsy here?

```suggestion
const element = document.getElementById(field.fieldId);
        
        if (element) {
          element.value = value;
        }
```

File: assets/src/js/gravityforms-client-side.js (hunk 2 above)
Comment (mleray) [reply assets/src/js/gravityforms-client-side.js:6]: No, because it's added at the top of the file to be used as global

**Comment by mardelnet** [comment]
> Update @mardelnet, here's Julia's answer:
> 
> _I think we don't need to worry for now, but I'll raise this with Molly to see if she can find out if these use cases are common... a solution would be to add an array to the DL event with form 1 values and then form 2 values, but my assumption is that it's very rare_

Should we create a follow-up ticket for this?

**Review by mardelnet — APPROVED** [review]

**Comment by mleray** [comment]
> Should we create a follow-up ticket for this?

@mardelnet I think it's fine for now, Molly and Julia will discuss it and I expect they'll open a ticket if needed.
